### ISSUE TITLE
Is this program compile ready ?

### ISSUE THREAD

**Issue by sandymanu** [Hidayathamir/telegram-forward-message-keyword#2]
I added the send and receiver id and compiled it. But it isn't running.

I get `decouple.UndefinedValueError: api_id not found. Declare it as envvar or define a default value` error.

I think the get-dialog is not imported correctly. Could you help ?

**Comment by Hidayathamir** [comment]
Hello brother, I just create readme, you can see it to know how to run this project, especially about `api_id` and `api_hash`.

**Comment by sandymanu** [comment]
Umm..It's still bit confusing to me even after going through Readme. I will study and see patiently.

I was basically looking for messages coming in one group with a particular keyword to be forwarded to another group. I have chat id's of both groups.

Here it looks like I have to create an app in my.telegram.org and use the api id and hash.

**Comment by Hidayathamir** [comment]
yeah thats right, just create an app in my.telegram.org. just fill in the form thought, its not hard.

**Comment by sandymanu** [comment]
Hi,

So i tried like you said, but the messages with keyword are not getting forwarded, despite having the keyword.

Initially the code didn't run as it was throwing the error `decouple.UndefinedValueError: 123456 not found. Declare it as envvar or define a default value.` (Note : 123456 is a random number I have entered here. The real code has actual api id)

To fix this error I directly entered the api number and hash, instead of via config code & then program runs fine.

The main.py code looks like this  https://pastebin.com/LzEjhUjq (Note : The api id, has keys and sender/receiver have entered random numbers in this paste for privacy]

Your thoughts on this would be appreciated.

**Comment by sandymanu** [comment]
https://github.com/voidbar/forwardgram/

Alternatively this code worked for me and it forwards message, but this one doesn't have the code for filtering by keyword. If you could help, it would be great !

**Comment by Hidayathamir** [comment]
Hai mate. here is the problem.
i'm using [python-decouple](https://pypi.org/project/python-decouple/) to hide my api_id and api_hash. Anyway i just update the readme, to show you how to config api_id and api_hash

**Comment by Hidayathamir** [comment]
according to your error `decouple.UndefinedValueError: 123456 not found. Declare it as envvar or define a default value.`
i guest you do it like this. 
`api_id = config("123456", cast=int)`
instead do like this.
`api_id = 123456`

**Comment by Hidayathamir** [comment]
[your code](https://pastebin.com/LzEjhUjq) is exactly how it should be

**Comment by Hidayathamir** [comment]
to add filter in [this code](https://github.com/voidbar/forwardgram/) you can do like this.

on line 38 after
`async def handler(event):`
you can check event.message to see is it contain keyword that you want. if true, then forward it.

it would be something like this
```
@client.on(events.NewMessage(chats=input_channels_entities))
async def handler(event):
    # Your logic here

    if 'keyword' in event.message.split():
        for output_channel in output_channel_entities:
            await client.forward_messages(output_channel, event.message)
```

**Comment by sandymanu** [comment]
> [your code](https://pastebin.com/LzEjhUjq) is exactly how it should be

Any idea why it doesn't work ?

**Comment by sandymanu** [comment]
>     # Your logic here

Not sure how to go with logic. Can you help out with the code ? You mean I should put out the logic in your code ?

**Comment by sandymanu** [comment]
If I add your code logic to that one, then the program starts and stops within a second. 

`2021-11-08 13:01:20,219 - root - INFO - Listening on 1 channels. Forwarding messages to 1 channels.`

It starts and stops within a second throwing the above log.


Below is the code.

https://pastebin.com/xViBTK2X

**Comment by Hidayathamir** [comment]
im sorry mate, i believe that project owner can help you

**Comment by sandymanu** [comment]
> im sorry mate, i believe that project owner can help you

Okay. Any idea why [your](https://pastebin.com/LzEjhUjq) code isn't working ? If I run your code, it isn't forwarding any messages.

**Comment by Hidayathamir** [comment]
this is what i suggest you do.
1. Check the senders id and receiver id again, run get-dialogs.py to find out the id.
2. Double-check your keywords, only messages containing keywords will be forwarded.
